src/systems/filecoin_mining/storage_mining/storage_mining_subsystem.go

```diff
@@ -383,14 +385,18 @@ func (sms *StorageMiningSubsystem_I) _submitSurprisePoStMessage(state stateTree.
 	params := make([]util.Serialization, 1)
 	params[0] = sector.Serialize_OnChainPoStVerifyInfo(sPoSt)
 
-	sysActor, ok := state.GetActor(addr.SystemActorAddr)
+	workerAddr, err := addr.Address_Make_Key(node_base.NETWORK, addr.KeyHash(sms._keyStore().WorkerKey().VRFPublicKey()))
+	if err != nil {
+		return err
+	}
```

Comment:
Alternatively it might be easier to read the worker address out of the MinerInfo on-chain structure and plumb it to here.

Reply:
yeah I thought about it but thought it easier to avoid reading the on-chain structures in order to get info about your own node...